code/game/machinery/vending.dm

```diff
@@ -521,20 +611,81 @@
 			if(!(R.category & src.categories))
 				return
 
-			if(R.price <= 0)
+			if(R.price <= 0 || !src.locked)
 				src.vend(R, usr)
 			else if(issilicon(usr)) //If the item is not free, provide feedback if a synth is trying to buy something.
 				usr << SPAN_DANGER("Artificial unit recognized.  Artificial units cannot complete this transaction.  Purchase canceled.")
 				return
 			else
 				src.currently_vending = R
-				if(!vendor_account || vendor_account.suspended)
+				if(!machine_vendor_account || machine_vendor_account.suspended)
 					src.status_message = "This machine is currently unable to process payments due to problems with the associated account."
 					src.status_error = 1
 				else
 					src.status_message = "Please swipe a card or insert cash to pay for the item."
 					src.status_error = 0
 
+		else if (href_list["setprice"] && !locked)
+			var/key = text2num(href_list["setprice"])
+			var/datum/data/vending_product/R = product_records[key]
+
+			R.price = input("Enter item price", "Item price") as num
+
+		else if (href_list["remove"] && !locked)
+			var/key = text2num(href_list["remove"])
+			var/datum/data/vending_product/R = product_records[key]
+
+			qdel(R)
+
+		else if (href_list["return"])
+			src.managing = FALSE
+
+		else if (href_list["management"])
+			src.managing = TRUE
+			src.status_message = "Welcome to the management screen."
+			src.status_error = 0
+
+		else if (href_list["setaccount"])
+			var/datum/money_account/newaccount = get_account(input("Please enter the number of the account that will receive profits from this Vendomat.", "Vendomat Account", null) as num)
+			if(!newaccount)
+				src.status_message = "No account specified. Vendomat will transfer its profits to its owner. Organization will be prioritized if set."
+				earnings_account = null
+			else
+				src.status_message = "This Vendomat will now transfer its profits to the specified account, owned by [newaccount.owner_name]."
+				src.status_error = 0
+				earnings_account = newaccount
+
+		else if (href_list["setdepartment"])
```

Comment:
This should call a proc to set the department instead of being done inline in the topic call.

In addition to this, we need the ability to set a compiletime var for a department, and have the vendor automatically set itself to that department when initialized